akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1058,6 +1059,33 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
     new Flow(delegate.mergeMat(that)(combinerToScala(matF)))
 
   /**
+   * Merge the given [[Source]] to this [[Flow]], taking elements as they arrive from input streams,
+   * picking always the smallest of the available elements (waiting for one element from each side
+   * to be available).
+   *
+   * '''Emits when''' one of the inputs has an element available
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' all upstreams complete
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def mergeSorted[U >: Out, M](that: Graph[SourceShape[U], M], comp: Comparator[U]): javadsl.Flow[In, U, Mat] =
```

Comment:
Well, not sure actually. The interpretation of “mergeSorted = merged by way of merge-sort” seems a bit far-fetched to me, while “merge, but sorted” is much shorter and closer at hand. And if the name does not reveal the semantics there’s always the documentation to refer to.

`mergePreSorted` does convey that something is pre-sorted here, but it does not convey the merge semantics themselves. `mergeSortedStreamsInSortedFashion` would be really precise, but people would just crucify us for picking such a name, I think ;-)